tests/tensorflow/test_tensorflow_estimator_export.py

```diff
@@ -0,0 +1,251 @@
+
+
+def test_iris_model_can_be_loaded_and_evaluated_successfully(tmpdir, model_path, monkeypatch):
+    monkeypatch.chdir(str(tmpdir))
+    model_data_info = save_or_log_tf_model_by_mlflow128(
+        str(tmpdir), "iris", "save_model", model_path
+    )
+
+    infer = mlflow.tensorflow.load_model(model_uri=model_path)
+    feed_dict = {
+        df_column_name: tf.constant(model_data_info.inference_df[df_column_name])
+        for df_column_name in list(model_data_info.inference_df)
+    }
+    raw_preds = infer(**feed_dict)
+    pred_dict = {column_name: raw_preds[column_name].numpy() for column_name in raw_preds.keys()}
+    for key in pred_dict.keys():
+        assert_array_almost_equal(pred_dict[key], model_data_info.raw_results[key])
+
+
+def test_log_and_load_model_persists_and_restores_model_successfully(tmpdir, monkeypatch):
+    monkeypatch.chdir(str(tmpdir))
+    model_data_info = save_or_log_tf_model_by_mlflow128(str(tmpdir), "iris", "log_model")
+    model_uri = f"runs:/{model_data_info.run_id}/model"
+    mlflow.tensorflow.load_model(model_uri=model_uri)
+
+
+def test_iris_data_model_can_be_loaded_and_evaluated_as_pyfunc(tmpdir, model_path, monkeypatch):
+    monkeypatch.chdir(str(tmpdir))
+    model_data_info = save_or_log_tf_model_by_mlflow128(
+        str(tmpdir), "iris", "save_model", model_path
+    )
+
+    pyfunc_wrapper = pyfunc.load_model(model_path)
+
+    # can call predict with a df
+    results_df = pyfunc_wrapper.predict(model_data_info.inference_df)
+    assert isinstance(results_df, pd.DataFrame)
+    assert_pandas_dataframe_almost_equal(results_df, model_data_info.raw_df)
+
+    # can also call predict with a dict
+    inp_dict = {}
+    for df_col_name in list(model_data_info.inference_df):
+        inp_dict[df_col_name] = model_data_info.inference_df[df_col_name].values
+    results = pyfunc_wrapper.predict(inp_dict)
+    assert isinstance(results, dict)
+    assert_pandas_dataframe_almost_equal(pd.DataFrame(results), model_data_info.raw_df)
+
+    # can not call predict with a list
+    inp_list = []
+    for df_col_name in list(model_data_info.inference_df):
+        inp_list.append(model_data_info.inference_df[df_col_name].values)
+    with pytest.raises(TypeError, match="Only dict and DataFrame input types are supported"):
+        results = pyfunc_wrapper.predict(inp_list)
+
+
+def test_categorical_model_can_be_loaded_and_evaluated_as_pyfunc(tmpdir, model_path, monkeypatch):
+    monkeypatch.chdir(str(tmpdir))
+    model_data_info = save_or_log_tf_model_by_mlflow128(
+        str(tmpdir), "categorical", "save_model", model_path
+    )
+
+    pyfunc_wrapper = pyfunc.load_model(model_path)
+
+    # can call predict with a df
+    results_df = pyfunc_wrapper.predict(model_data_info.inference_df)
+    # Precision is less accurate for the categorical model when we load back the saved model.
+    pandas.testing.assert_frame_equal(
+        results_df, model_data_info.expected_results_df, check_less_precise=3
```

Comment:
Let's `rtol` and `atol`. `check_less_precise` is deprecated.

https://pandas.pydata.org/docs/reference/api/pandas.testing.assert_frame_equal.html